### PR TITLE
Remove SET_FOCUS message and call setFocus() on tab switches.

### PR DESCRIPTION
**Pull request by JuBan1**
There's a regression with document focus that's in this branch and in current master.

Open two documents side-by-side. Switching the active editor by clicking on the tab bar doesn't move the active focus (aka the blinking caret) to the clicked tab.

That seems properly fixed by calling `setFocus` in the `onEditorChanged()` callback.

Then I also removed the async stuff called during focus events. I couldn't find anything wrong with this. Can you check it out?

### CODE REVIEW

**Comment by tekladis** [comment]
Everything appears to work here.  I'll go ahead and merge it.  I've got no clue at all when this will get merged to master, however.  Daniel has almost dropped off of the face of the earth and it's been that way for almost a month.  I know he was talking about stagnating code, but it definitely becomes stagnant if no code is being added.

I'm not really sure what to do at this point.  There's a lot of good changes that I'd like to see made that are coming quite a bit slower due to lack of participation, conflicts of interest, and simple policies revolving around commits (i.e. keeping commits super small can be a real hassle when refactoring code).

I really do hate the thought of a possible fork, but OTE is just going to go to waste with how things are going at the moment.  There's also the issue of updating the build system and rearranging the file structure which won't be able to happen until v2.0 which requires QWebEngine to be... stable?

Don't mind me, I'm just having a bit of a rant here due to the issue list growing uncontrollably with no real way to fix anything. :|

**Comment by JuBan1** [comment]
We both vetted it, I'll merge it. I don't use it much when Daniel is around but him being unavailable was the reason I got merge powers to begin with.

There is also a growing number of issues that are hard to solve with CodeMirror due to their nature (disabling text dragging, double-click highlighting not working with accented characters, undoing regex replace, etc).

I've been in a bit of a slump since hearing that OTE won't be merged into Nqq after all. It's a lot less interesting when I'm doing the work only for myself. 

I don't really want to try to split the community by packaging and distributing the OTE branch separately but I don't know how else to achieve that maturity Daniel is looking for.

**Comment by tekladis** [comment]
I'm right there with you.  The entire reason my PRs slowed down so much was to wait for OTE integration.  The fact of the matter is between the hard-to-fix bugs and poor usage of CPU/memory I don't find NQQ as interesting as it once was.  

There was a time.... well before you arrived where NQQ used QScintilla.  As far as I'm aware it was dropped in favor of CodeMirror to try and help increase production.  There were many upsides to the switch at the time as new syntax highlighting rules were an extreme pain to implement and the API itself was older than father time.  It's what I'd consider to be legacy code.

With that said, you can only squeeze so much juice out of javascript before it begins to falter.  This is especially true on older systems.  I've never had a ton of money so most of my development performance-wise targets the lowest common denominator at that time, which is usually me.  My systems upgrades tend to be 5 years or so apart.

Back when I was first learning to program the performance of software mattered quite a bit more than it does in today's world, which is probably why I'm so critical about the entire situation.  I'm not a performance masochist by any means but I don't think I'm wrong to believe that a text editor taking up more RAM than a good quality 2d game is bad.

When it boils down to it I never got into the "javascript as an application" scene.  The idea itself is interesting but in a majority of cases it produces software that either doesn't integrate well with the system, performs poorly, or both.

TL;DR I don't want to split the community either, but creative differences appear to be showing its head.  I tend to lean toward whichever method will produce results.  A fork tends to take a little time to pick up steam, but having a less restricted workflow, even temporarily, would really help fuel some nice progress.  The original goal of the project was actually to mimic notepad++ in features, but there's quite a few that are either hard or downright impossible (for me) to implement using javascript.

**Comment by tekladis** [comment]
@JuBan1 if you're committing to doing a fork, I might recommend actually enabling the issues tracker on your repository so bugs can be reported, lol.  I'll be over shortly to start helping put things together.  

Depending on how much you're wanting to diverge from the base it'd probably be a good time to make updates to the file structure and such as well so it's easier to navigate the source for newcomers.  NQQ used a very non-standard file system structure from the get go and ended up staying that way for a good while mainly due to it being overlooked.

Actually, the list of issues that can be fixed in a fork is fairly large, this also includes long-standing feature requests that have been overlooked for a long long time.

**Comment by JuBan1** [comment]
Oh my. I didn't notice issues were turned off.

I kinda-sorta set up [this branch](https://github.com/JuBan1/notepadqq/tree/ote) to be a proper fork, with changed install scripts and README. I just made it the default one too.

About the changes: The goal is still to eventually merge OTE into upstream Nqq. I can't stop the diff between those two branches from growing larger over time but we shouldn't actively encourage that if possible. So I think we should restrict Nqq-OTE to changes that concern OTE and try to introduce general changes to upstream Nqq.

Changing file structure (which I'd appreciate too) is something that can be done in upstream and then pulled into Nqq-OTE.

I'm open to suggestions but I think that's the best way forward.

**Comment by JuBan1** [comment]
Okay, I've added a bit of an overview [into the issues tab](https://github.com/JuBan1/notepadqq/issues) as to how things are going.
